File: source_code/functional_result_calc.cpp
static std::expected<int, std::string_view>
perform_calculation(int input) {
    if (input % 2 != 0) {
        return std::unexpected{
            "Input must be odd"
        };
    }
    return input * 3;
}

static void
use_expected(int input) {
    auto res = perform_calculation(input);
    if (res) {
        std::print("{}", *res);
    } else {
        std::print("Err: {}",
            res.error());
    }
}

File: source_code/functional_result_calc.rs
fn perform_calculation
        (input: i32)
        -> Result<i32, &'static str> {
    if input % 2 != 0 {
        Err("Input must be odd")
    } else {
        Ok(input * 3)
    }
}

fn use_result(input: i32) {
    match perform_calculation(input) {
        Ok(res)
            => println!("{}", res),
        Err(msg)
            => println!("Err: {}", msg),
    }
}